async-stripe/src/hyper/client.rs

```
@@ -109,7 +109,9 @@ impl Client {
         let mut last_error = StripeError::ClientError("invalid strategy".into());
 
         if let Some(key) = strategy.get_key() {
-            req_builder = req_builder.header(HeaderName::from_static("Idempotency-Key"), key);
+            const HEADER_NAME: HeaderName = HeaderName::from_static("idempotency-key");
+            assert!(!key.is_empty(), "idempotency key is empty");
```

Probably would be better to return an error here, could add a variant to `StripeError`. Even better would be to validate on construction of `RequestStrategy`, but that's not currently possible without hiding more of the internal representation there.

---

yeah, I put that assertion here because I wanted to narrow down the out of bound index and then I left it there because it seemed a good sanity check.
However I also think that having a proper error would be better, so I created a commit that adds a newtype for the idempotency key. It is basically a string that cannot be empty or longer than 255 chars. It has its own error, because I think its mostly isolated from the rest of the logic. Having it mostly transparent struct also doesn't hide any implementation and representation